ml-agents/mlagents/trainers/torch/layers.py

```diff
@@ -82,3 +84,62 @@ def lstm_layer(
                         forget_bias
                     )
     return lstm
+
+
+class MemoryModule(torch.nn.Module):
+    @abc.abstractproperty
+    def memory_size(self) -> int:
+        """
+        Size of memory that is required at the start of a sequence.
+        """
+        pass
+
+    @abc.abstractmethod
+    def forward(
+        self, input_tensor: torch.Tensor, memories: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Pass a sequence to the memory module.
+        :input_tensor: Tensor of shape (batch_size, seq_length, size) that represents the input.
+        :memories: Tensor of initial memories.
+        :return: Tuple of output, final memories.
+        """
+        pass
+
+
+class LSTM(MemoryModule):
```

Comment:
Maybe add a test specific for this Module. (unless it is already tested enough in the network_tests)

Reply:
It's tested in both the network tests and the lstm_layer is tested in the layers test, but I can add another test

Reply:
Added